core/src/main/java/com/google/errorprone/bugpatterns/TypeParameterNaming.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.Category.JDK;
+import static com.google.errorprone.BugPattern.SeverityLevel.SUGGESTION;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Ascii;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Streams;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.LinkType;
+import com.google.errorprone.BugPattern.ProvidesFix;
+import com.google.errorprone.BugPattern.StandardTags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.TypeParameterTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.names.NamingConventions;
+import com.sun.source.tree.TypeParameterTree;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.Name;
+
+/** Enforces type parameters match the google style guide. */
+@BugPattern(
+  name = "TypeParameterNaming",
+  summary =
+      "Type Parameters must be a single letter with a numeric suffix,"
```

Comment:
@nglorioso @ronshapiro I think that this portion of the sentence is a bit misleading, as it seems to suggest that the numeric suffix is mandatory and can be of any length, whereas the guide seems to say the numeric suffix is optional and should only be one digit long if used. What do you both think?

Reply:
I didn't write it (just synced) but I think this should read:
```java
"Type Parameters must be a single letter with an optional numeric suffix"
```

Reply:
@ronshapiro Sounds a very good idea to me! I'd be perfectly happy with that rewording.